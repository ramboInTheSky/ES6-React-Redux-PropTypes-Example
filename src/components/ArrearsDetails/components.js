import styled from 'styled-components';
import { Button, Typography, mediaQueries } from 'nhh-styles';

export const Heading = styled(Typography.H3)`
  margin-bottom: 1.5rem;
`;

export const Label = styled.span`
  font-weight: 700;
  ${mediaQueries.desktop`
    white-space: nowrap;
  `};
  ${mediaQueries.tablet`
    white-space: nowrap;
  `};
`;

export const LabelLink = styled(Button)`
  font-weight: 700;
  font-size: 18px;
  padding: 0;
  text-align: left;
`;

export const Wrapper = styled.div`
  margin-bottom: 1.5rem;
`;
