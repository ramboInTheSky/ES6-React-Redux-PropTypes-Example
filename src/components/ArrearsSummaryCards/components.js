import styled from 'styled-components';
import { mediaQueries } from 'nhh-styles';

// eslint-disable-next-line import/prefer-default-export
export const Wrapper = styled.div`
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  margin-top: 12px;

  > a {
    color: inherit;
    margin-bottom: 16px;
    margin-right: 1%;
    width: 32%;

    &:hover {
      color: inherit;
      text-decoration: none;
    }

    ${mediaQueries.mobile`
      margin-right: 0;
      width: 100%;
    `};

    ${mediaQueries.tablet`
      width: 49%;
    `};
  }
`;
