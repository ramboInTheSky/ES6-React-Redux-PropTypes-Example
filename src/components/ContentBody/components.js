import styled from 'styled-components';

export const OuterContainer = styled.div`
  padding-top: 50px;
  padding-bottom: 50px;
`;
export const InnerContainer = styled.div``;
