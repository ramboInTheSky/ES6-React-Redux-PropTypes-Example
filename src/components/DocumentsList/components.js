import styled from 'styled-components';
import { Button, mediaQueries, Typography } from 'nhh-styles';

export const H3 = styled(Typography.H3)`
  margin-bottom: 30px;
`;

export const Link = styled(Button)`
  font-size: inherit;
  font-weight: 400;
  display: inline;
`;

export const Container = styled.section`
  padding: 25px 28px 0;

  ${mediaQueries.mobile`
    padding: 15px 18px;
  `};
`;

export const ErrorMessage = styled.div`
  color: ${({ theme }) => (theme ? theme.colors.support.two : 'inherit')};
  margin-bottom: 50px;
`;

export const DateField = styled.div``;

export const Row = styled.div`
  border-bottom: 1px solid ${({ theme }) => theme.colors.tertiaryLighter};
  color: ${({ theme }) => theme.colors.tertiary};
  display: flex;
  flex-direction: row;
  font-size: 16px;
  padding: 8px 0;
`;

export const Header = Row.extend`
  font-size: 14px;
  font-weight: bold;
  text-transform: uppercase;
`;

export const Column = styled.div`
  width: ${({ width }) => width || 'auto'};
  text-align: ${({ align }) => align || 'left'};
  padding-right: 8px;
  word-wrap: break-word;

  &:last-child {
    padding-right: 0;
  }

  ${mediaQueries.mobile`
    padding-right: 4px;
  `};
`;

export const DesktopView = styled.div`
  display: none;

  ${mediaQueries.desktop`
    display: block;
  `};
`;

export const MobileAndTabletView = styled.div`
  display: block;

  ${mediaQueries.desktop`
    display: none;
  `};
`;
