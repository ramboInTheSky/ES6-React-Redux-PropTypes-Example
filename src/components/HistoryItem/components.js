import styled from 'styled-components';
import { Typography } from 'nhh-styles';

export const Label = styled.span`
  font-weight: 700;
`;

export const Title = styled(Typography.H3)`
  margin: -30px 0 30px 0;
`;

export const Value = styled.span`
  word-break: break-word;
`;
