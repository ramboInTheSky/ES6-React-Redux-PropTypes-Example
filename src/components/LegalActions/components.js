import styled, { withTheme } from 'styled-components';
import { Typography } from 'nhh-styles';

export const Wrapper = styled.div`
  margin: 1.5rem 0 1.875rem 0;
`;

export const P = styled(Typography.P)`
  margin: 1rem 0;
`;

export const Label = styled.span`
  font-weight: 700;
`;

export const Highlight = withTheme(styled.span`
  color: ${({ theme, isExpiring }) => (isExpiring ? theme.colors.support.two : 'inherit')};
  ${({ isExpiring }) => isExpiring && `font-weight: 700;`};
`);
