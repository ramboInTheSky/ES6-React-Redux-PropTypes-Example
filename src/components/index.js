export { default as ArrearsDetails } from './ArrearsDetails';
export { default as ArrearsSummaryCards } from './ArrearsSummaryCards';
export { default as ButtonRow } from './ButtonRow';
export { default as ContentBody } from './ContentBody';
export { default as ErrorBoundary } from './ErrorBoundary';
export { default as FieldRow } from './FieldRow';
export { default as FormWrapper } from './FormWrapper';
export { default as Heading } from './Heading';
export { default as HistoryItem } from './HistoryItem';
export { default as LegalActions } from './LegalActions';
export { default as MediaTableView } from './MediaTableView';
export { default as ModalFieldRow } from './ModalFieldRow';
export { default as NoTouching } from './NoTouching';
export { default as PreviewLinkContainer } from './PreviewLinkContainer';
export { default as PreviewTemplate } from './PreviewTemplate';
export { default as PreviewLink } from './PreviewLink';
export { default as NotificationWrapper } from './NotificationWrapper';
export { default as TaskTitle } from './TaskTitle';
export { default as TasksSummaryCards } from './TasksSummaryCards';
export { default as TemplateContainer } from './TemplateContainer';
