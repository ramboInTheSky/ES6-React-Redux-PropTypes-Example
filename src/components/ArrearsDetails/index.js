import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import format from 'string-format';
import { formatting, Table, tap } from 'nhh-styles';
import has from 'ramda/src/has';

import { getPauseDetails, getPaymentPlanDetails } from './selectors';
import { Label, LabelLink, Wrapper } from './components';
import { Heading } from '../';
import getInternalRoutes from '../../constants/internalRoutes';
import { DRAFT, ACTIVE } from '../../constants/pause';

const getTableData = (labels, props) => {
  const {
    arrearsLength,
    createdOn,
    owner,
    pauseSummary,
    paymentPlan,
    referralSummaries,
    rentDetail,
    status,
    workwiseReferenceNumber,
  } = props;
  const internalRouter = getInternalRoutes({ ...props });

  const PaymentPlanLabel = paymentPlan ? LabelLink : Label;
  const pause = pauseSummary.find(item => item.status === ACTIVE);
  const draftPause = pauseSummary.find(item => item.status === DRAFT);
  const PauseLabel = pause ? LabelLink : Label;
  const DraftPauseLabel = draftPause ? LabelLink : Label;
  const isReferralSummaries = referralSummaries && referralSummaries.length;
  const latestReferralSummary =
    isReferralSummaries && referralSummaries[referralSummaries.length - 1];

  const hasRentProp = property =>
    tap(rentDetail && has(property, rentDetail), () => rentDetail[property]);

  const returnArray = [
    [
      <Label>{labels.patchOwner}</Label>,
      <span data-bdd="ArrearsDetails-patchOwner">{owner.name}</span>,
    ],
    [<Label>{labels.status}</Label>, <span data-bdd="ArrearsDetails-status">{status}</span>],
    [
      <Label>{labels.workwiseReference}</Label>,
      <span data-bdd="ArrearsDetails-workwiseReference">{workwiseReferenceNumber}</span>,
    ],
    [
      <Label>{labels.lengthOfArrears}</Label>,
      <span data-bdd="ArrearsDetails-lengthOfArrears">
        {format(labels.lengthOfArrearsValue, {
          days: arrearsLength,
          date: formatting.formatDate(createdOn),
        })}
      </span>,
    ],
    [
      <Label>{labels.rentDetails.paymentReferenceNumber}</Label>,
      <span data-bdd="ArrearsDetails-paymentReferenceNumber">
        {hasRentProp('paymentReference') || labels.none}
      </span>,
    ],
    [
      <Label>
        {format(labels.rentDetails.collectionRate.label, {
          value: labels.rentDetails.collectionRate.value5w,
        })}
      </Label>,
      <span data-bdd="ArrearsDetails-paymentCollectionRate5w">
        {tap(hasRentProp('collectionRate5Weeks'), val =>
          format(labels.rentDetails.collectionRate.value, {
            value: val,
          })
        ) || labels.none}
      </span>,
    ],
    [
      <Label>
        {format(labels.rentDetails.collectionRate.label, {
          value: labels.rentDetails.collectionRate.value4m,
        })}
      </Label>,
      <span data-bdd="ArrearsDetails-paymentCollectionRate4m">
        {tap(hasRentProp('collectionRate4Months'), val =>
          format(labels.rentDetails.collectionRate.value, {
            value: val,
          })
        ) || labels.none}
      </span>,
    ],
    [
      <Label>{labels.rentDetails.grossRent.label}</Label>,
      <span data-bdd="ArrearsDetails-grossPayment">
        {tap(hasRentProp('grossRent'), formatting.formatCurrency) || labels.none}
      </span>,
    ],
    ...(tap(hasRentProp('charges'), charges =>
      charges.map(charge => [
        <Label>{charge.description}</Label>,
        <span data-bdd={`ArrearsDetails-charge-${charge.description}`}>
          {formatting.formatCurrency(charge.chargeValue)}
        </span>,
      ])
    ) || []),
    [
      <Label>{labels.rentDetails.paymentFrequency}</Label>,
      <span data-bdd="ArrearsDetails-paymentReferenceNumber">
        {rentDetail ? rentDetail.paymentFrequency : labels.none}
      </span>,
    ],
    [
      <Label>{labels.rentDetails.arrearsBalance}</Label>,
      <span data-bdd="ArrearsDetails-arrearsCurrentBalance">
        {tap(hasRentProp('currentBalance'), val =>
          formatting.formatCurrency(val === 0 ? 0 : val * -1)
        ) || labels.none}
      </span>,
    ],
    [
      <PaymentPlanLabel isText to={internalRouter.paymentPlan}>
        {labels.paymentPlan.label}
      </PaymentPlanLabel>,
      <span data-bdd="ArrearsDetails-paymentPlan">
        {paymentPlan ? getPaymentPlanDetails(paymentPlan, labels) : labels.none}
      </span>,
    ],
    [
      <PauseLabel isText to={internalRouter.pause}>
        {labels.pauses.label}
      </PauseLabel>,
      <span data-bdd="ArrearsDetails-pauses">{getPauseDetails(pause, labels)}</span>,
    ],
    [
      <DraftPauseLabel isText to={internalRouter.pause}>
        {labels.pauses.draftLabel}
      </DraftPauseLabel>,
      <span data-bdd="ArrearsDetails-draftPauses">
        {getPauseDetails(draftPause, labels, 'draft')}
      </span>,
    ],
    [
      <Label>{labels.referralStatus.label}</Label>,
      <span data-bdd="ArrearsDetails-referralStatus">
        {latestReferralSummary ? (
          <Fragment key={`${latestReferralSummary.teamName}_${latestReferralSummary.createdOn}`}>
            {format(labels.referralStatus.value, {
              team: latestReferralSummary.teamName,
              date: formatting.formatDate(latestReferralSummary.createdOn),
            })}
            <br />
          </Fragment>
        ) : (
          labels.none
        )}
      </span>,
    ],
  ];

  return returnArray;
};

const ArrearsDetails = ({ heading, labels, ...props }) => (
  <Wrapper>
    <Heading>{heading}</Heading>
    <Table
      addExtraTd={false}
      addExtraTh={false}
      data-bdd="ArrearsDetailsTable"
      data={getTableData(labels, props)}
      striped
      tdWrap
      topAlign
    />
  </Wrapper>
);

export const detailsPropTypes = {
  createdOn: PropTypes.string.isRequired,
  owner: PropTypes.shape({
    name: PropTypes.string.isRequired,
    id: PropTypes.string,
    type: PropTypes.string,
  }),
  pauseSummary: PropTypes.arrayOf(
    PropTypes.shape({
      expiryDate: PropTypes.string.isRequired,
      startDate: PropTypes.string.isRequired,
      status: PropTypes.string.isRequired,
    })
  ),
  paymentPlan: PropTypes.shape({
    endDate: PropTypes.string.isRequired,
    installment: PropTypes.shape({
      amount: PropTypes.number.isRequired,
      period: PropTypes.number.isRequired,
      schedule: PropTypes.string.isRequired,
    }).isRequired,
    startDate: PropTypes.string.isRequired,
    type: PropTypes.string.isRequired,
  }),
  referralSummaries: PropTypes.arrayOf(
    PropTypes.shape({
      createdOn: PropTypes.string.isRequired,
      teamName: PropTypes.string.isRequired,
    })
  ),
  status: PropTypes.string.isRequired,
  arrearsLength: PropTypes.number,
  workwiseReferenceNumber: PropTypes.string,
};

export const labelPropTypes = {
  lengthOfArrears: PropTypes.string.isRequired,
  none: PropTypes.string.isRequired,
  patchOwner: PropTypes.string.isRequired,
  pauses: PropTypes.shape({
    active: PropTypes.string.isRequired,
    draftLabel: PropTypes.string.isRequired,
    draftPausePeriod: PropTypes.string.isRequired,
    label: PropTypes.string.isRequired,
    pausePeriod: PropTypes.string.isRequired,
    pendingApproval: PropTypes.string.isRequired,
  }).isRequired,
  paymentPlan: PropTypes.shape({
    endDate: PropTypes.string.isRequired,
    frequency: PropTypes.string.isRequired,
    instalmentAmount: PropTypes.string.isRequired,
    label: PropTypes.string.isRequired,
    period: PropTypes.string.isRequired,
    startDate: PropTypes.string.isRequired,
    type: PropTypes.string.isRequired,
  }).isRequired,
  referralStatus: PropTypes.shape({
    label: PropTypes.string.isRequired,
    value: PropTypes.string.isRequired,
  }).isRequired,
  status: PropTypes.string.isRequired,
  workwiseReference: PropTypes.string.isRequired,
};

ArrearsDetails.defaultProps = {
  paymentPlan: null,
  pauseSummary: [],
};

ArrearsDetails.propTypes = {
  ...detailsPropTypes,
  heading: PropTypes.string.isRequired,
  labels: PropTypes.shape(labelPropTypes).isRequired,
};

export default ArrearsDetails;
