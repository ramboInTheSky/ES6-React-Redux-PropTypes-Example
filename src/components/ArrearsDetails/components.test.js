import React from 'react';
import { shallow } from 'enzyme';

import 'jest-styled-components';

import { Heading, Label, LabelLink, Wrapper } from './components';

describe('ArrearsDetails components', () => {
  it('should render <Heading /> correctly', () => {
    expect(shallow(<Heading />)).toMatchSnapshot();
  });

  it('should render <Label /> correctly', () => {
    expect(shallow(<Label />)).toMatchSnapshot();
  });

  it('should render <LabelLink /> correctly', () => {
    expect(shallow(<LabelLink />)).toMatchSnapshot();
  });

  it('should render <Wrapper /> correctly', () => {
    expect(shallow(<Wrapper />)).toMatchSnapshot();
  });
});
