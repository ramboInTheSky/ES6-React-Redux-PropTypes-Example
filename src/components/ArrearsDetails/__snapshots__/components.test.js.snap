// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArrearsDetails components should render <Heading /> correctly 1`] = `
.c0 {
  margin-bottom: 1.5rem;
}

<styled.h3
  className="c0"
/>
`;

exports[`ArrearsDetails components should render <Label /> correctly 1`] = `
.c0 {
  font-weight: 700;
}

@media (min-width:1001px) {
  .c0 {
    white-space: nowrap;
  }
}

@media (min-width:768px) and (max-width:1000px) {
  .c0 {
    white-space: nowrap;
  }
}

<span
  className="c0"
/>
`;

exports[`ArrearsDetails components should render <LabelLink /> correctly 1`] = `
.c0 {
  font-weight: 700;
  font-size: 18px;
  padding: 0;
  text-align: left;
}

<WithTheme(Button)
  className="c0"
/>
`;

exports[`ArrearsDetails components should render <Wrapper /> correctly 1`] = `
.c0 {
  margin-bottom: 1.5rem;
}

<div
  className="c0"
/>
`;
